### DISCUSSION TITLE
watch current Pattern while the song is running Hydrogen 1.1.0

### DISCUSSION THREAD

**Discussion by maurobiamino74** [hydrogen-music/hydrogen#1409] [Hydrogen on OS X]
Good morning to every one,
When I play the song with some patterns, I can't watch the current pattern which is playing.
I can watch it only if I select it manually.
I can't believe it as Hydrogen has a lot of functions I think it has to have it, it would be very useful to study and practice .

Thank you if somebody will answer to me.

have a nice one , Mauro.

**Comment by cme** [comment]
This is something I've thought about as well, for more or less the same reasons. The trouble is that, in the general case, there is no single "current" pattern -- multiple patterns can be played simultaneously.

I thought about addressing this in the context of improvements to Stacked Pattern mode (see #1310 ) , perhaps by showing all currently playing patterns in the same way.

Closely related, there's #717 which has the same problem -- there's no single "current" pattern.

**Reply by theGreatWhiteShark** [reply]
> This is something I've thought about as well, for more or less the same reasons. The trouble is that, in the general case, there is no single "current" pattern -- multiple patterns can be played simultaneously.

Yeah, I thought about this too. And I think we should just do it.

In Stacked Pattern mode it's basically the same issue with recording and we already addressed this as well (notes go to the bottom-most pattern, right?). With this feature activated the song editor would always show the pattern recorded notes will be inserted to.

But the feature might be annoying if it can only be set via the preferences. How about fusing the Select and Draw mode into a single button and reusing one of them for this feature? Any idea about it's name or icon? (maybe "follow pattern")

> I thought about addressing this in the context of improvements to Stacked Pattern mode (see #1310 ) , perhaps by showing all currently playing patterns in the same way.

This would still be a great improvement.

**Reply by maurobiamino74** [reply]
Thank you for your informations, but  my request is easier for the way I use Hydrogen;
Normally I make different patterns, then when I select play song they play on by one in sequence as all the demos which are in project window at the up main bar,  I can't watch the pattern is running , I watch that only  if I select it manually .
To explain better what I would have: the pattern which is playing, is when I watch it while song is running and the arrow runs up it with the current speed beat.
I can't understand because there isn't that function in a Great program as this, for me is many important to watch  alive what pattern/s  is/are playing; for example I need it  when  I'm playing a true drum to exercise myself. That function would run like music sheet.
Maybe am I that  I can't  understand because this program maybe hasn't been made to use how I have asked.

Sorry again:  are you thinking about to advise me, that  I would have to put this question on "Issues" section, by the way?

Thank you, Mauro.

**Reply by theGreatWhiteShark** [reply]
> Maybe am I that I can't understand because this program maybe hasn't been made to use how I have asked.

There are various ways to use it and your's is the one I use too. Another way would be the stacked pattern mode (see https://github.com/hydrogen-music/hydrogen/pull/1310 for an example/image). 

> I can't understand because there isn't that function

Well, it grew quite naturally and there are a number of missing peaces in the UX. But we are doing our best to close gaps.

> Sorry again: are you thinking about to advise me, that I would have to put this question on "Issues" section, by the way?

That's indeed the better place for a feature request. But don't worry.

**Reply by theGreatWhiteShark** [reply]
This was implemented in #1563
